emulator/blockchain.go

```diff
@@ -1629,6 +1645,16 @@ func (b *Blockchain) SetClock(clock Clock) {
 	b.pendingBlock.SetClock(clock)
 }
 
+// NewScriptEnvironment returns an environment.Environment by
+// using as a storage snapshot the blockchain's ledger state.
+// Useful for tools that use the emulator's blockchain as a library.
+func (b *Blockchain) NewScriptEnvironment() environment.Environment {
+	return environment.NewScriptEnvironmentFromStorageSnapshot(
+		b.vmCtx.EnvironmentParams,
+		b.pendingBlock.ledgerState.NewChild(),
```

Comment:
Were these taken/copied from another place? Just curious, don't really know if that's how `NewScriptEnvironmentFromStorageSnapshot` is supposed to be initialized

Reply:
Yes, I got them from the `Blockchain.GetSourceFile` method, so I have updated it to use this new helper function in: https://github.com/onflow/flow-emulator/pull/476/commits/e300dd2799807742587424072a915af36f44aecc